### DISCUSSION TITLE
how do I expose swagger docs without requiring a JWT bearer token for authorization

### DISCUSSION THREAD

**Discussion by tonykgill** [aws-powertools/powertools-lambda-python#3584] [Q&A]
Hi folks,

I'm new to aws lambda powertools for python, and relatively new to Lambda in general. I really like powertools. A recently added feature allows us to create a swagger endpoint for docs. This is great.

But... our APIs require a JWT bearer token for authorization. What's the recommended method for exposing the docs without requiring the JWT token?

The same point was raised [here](https://github.com/aws-powertools/powertools-lambda-python/issues/2421#issuecomment-1759475248). But I can't find a resolution as yet across the other issues and discussions.

Thanks

**Comment by tonykgill** [comment]
I have a solution. I can create a new /swagger endpoint in my API Gateway, which doesn't require authentication, and integrate it with a different Lambda function. Calls to the /swagger endpoint show the docs. Of course the 'try it' button fails without auth. But I think I have what I need for now, which was just to expose the docs.

**Reply by leandrodamascena** [reply]
Hello @tonykgill! Today was a very busy day and I couldn't reply here.

You're right, it goes straight to what you thought: an endpoint without authentication.

Tomorrow morning I'll left some more considerations and explain how it works in API Gateway.

Thanks

**Reply by tonykgill** [reply]
Awesome. Thank you @leandrodamascena.
